src/sentry/static/sentry/app/components/group/externalIssueActions.jsx

```diff
@@ -151,114 +153,189 @@ class ExternalIssueActions extends AsyncComponent {
   }
 }
 
-export class SentryAppExternalIssueActions extends React.Component {
-  static propTypes = {
-    group: PropTypes.object.isRequired,
-    sentryAppComponent: PropTypes.object.isRequired,
-    sentryAppInstallation: PropTypes.object,
-    externalIssue: PropTypes.object,
-  };
+export const SentryAppExternalIssueActions = withApi(
+  class SentryAppExternalIssueActions extends React.Component {
+    static propTypes = {
+      api: PropTypes.object.isRequired,
+      group: PropTypes.object.isRequired,
+      sentryAppComponent: PropTypes.object.isRequired,
+      sentryAppInstallation: PropTypes.object,
+      externalIssue: PropTypes.object,
+    };
 
-  constructor(props) {
-    super(props);
+    constructor(props) {
+      super(props);
 
-    this.state = {
-      action: 'create',
-      showModal: false,
+      this.state = {
+        action: 'create',
+        externalIssue: props.externalIssue,
+        showModal: false,
+      };
+    }
+
+    componentWillUpdate(prevProps) {
```

Comment:
This method is deprecated you can generally use `componentDidUpdate` as a replacement.

Reply:
Yea, I remember that, but then our linter complains about using `setState` in `componentDidUpdate` so I'm not sure what to do, tbh.

Reply:
You can move the setState() call into a separate method to trick the linter. I find that linter warning generally not helpful. Perhaps it should be disabled?

Reply:
Heh, that works. Yea, it seems like it was a somewhat recent addition? I feel like I've definitely used `setState` in there recently without a warning. 🤷‍♂️

Reply:
+1 for disabling